server/src/BitBakeProjectScanner.ts

```diff
@@ -400,68 +385,56 @@ export class BitBakeProjectScanner {
     }
   }
 
-  private executeCommandInBitBakeEnvironment (command: string, machine: string | undefined = undefined): string {
-    let returnValue: string = ''
-
-    if (this.isBitbakeAvailable()) {
-      const scriptContent: string = this.generateBitBakeCommandScriptFileContent(command, machine)
-      const pathToScriptFile: string = this._projectPath + '/' + this._settingsWorkingFolder
-      const scriptFileName: string = pathToScriptFile + '/executeBitBakeCmd.sh'
+  private executeCommandInBitBakeEnvironment (command: string): string {
+    const scriptContent: string = this.generateBitBakeCommandScriptFileContent(command)
+    const pathToScriptFile: string = this._projectPath + '/' + this._pathToBuildFolder
+    const scriptFileName: string = pathToScriptFile + '/executeBitBakeCmd.sh'
 
-      if (!fs.existsSync(pathToScriptFile)) {
-        fs.mkdirSync(pathToScriptFile)
-      }
-      fs.writeFileSync(scriptFileName, scriptContent)
-      fs.chmodSync(scriptFileName, '0755')
-
-      returnValue = this.executeCommand(scriptFileName)
+    if (!fs.existsSync(pathToScriptFile)) {
+      fs.mkdirSync(pathToScriptFile)
     }
+    fs.writeFileSync(scriptFileName, scriptContent)
+    fs.chmodSync(scriptFileName, '0755')
 
-    return returnValue
+    return this.executeCommand(scriptFileName)
   }
 
   private executeCommand (command: string): string {
     let stdOutput: string = ''
-
-    if (this._projectPath !== null) {
-      const returnObject = execa.shellSync(command)
-
-      if (returnObject.status === 0) {
-        stdOutput = returnObject.stdout
-      } else {
-        const data: Buffer = fs.readFileSync(command)
-        logger.error('error on executing command: ' + data.toString())
+    try {
+      if (this._projectPath !== null) {
+        const returnObject = execa.shellSync(command)
+        if (returnObject.status === 0) {
+          stdOutput = returnObject.stdout
+        } else {
+          const data: Buffer = fs.readFileSync(command)
+          logger.error('error on executing command: ' + data.toString())
+        }
       }
+    } catch (error: any) {
```

Comment:
I'm not sure we should catch the error here because we don't know how to handle it. All callers will also have to check for errors. Error handling should be done at the calling level that can recover, or abort and display an error message.

Reply:
This code is outdated, it does not exist anymore